disnake/message.py

```diff
@@ -153,7 +153,13 @@ async def _edit_handler(
     if embed is not MISSING:
         embeds = [embed] if embed else []
     if embeds is not MISSING:
-        payload["embeds"] = [e.to_dict() for e in embeds]
+        _embeds = []
+        for e in embeds:
+            if e._files:
+                e._files.clear()
+            _embeds.append(e.to_dict())
+
+        payload["embeds"] = _embeds
         for embed in embeds:
             if embed._files:
                 files = files or []
```

Comment:
This won't work for uploading a new embed on edit; perhaps this could instead only add files from `embed._files` to `files` if they aren't closed yet?

Reply:
yeah should work